### PR TITLE
Detect some errors in port name, such as 'http://address/'

### PR DESCRIPTION
**Pull request by bboreham**
To fix https://github.com/weaveworks/weave/issues/1892

### CODE REVIEW

File: connection_maker.go [hunk 1]
```diff
@@ -100,7 +101,9 @@ func (cm *ConnectionMaker) InitiateConnections(peers []string, replace bool) []e
 			host = peer
 			port = "0" // we use that as an indication that "no port was supplied"
 		}
-		if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
+		if !isAlnum(port) {
+			errors = append(errors, fmt.Errorf("invalid peer name '%s', should just be host[:port]", peer))
```
Comment (peterbourgon) [inline connection_maker.go]: `'%s'` is better pronounced `%q`

File: connection_maker.go [hunk 2]
```diff
@@ -100,7 +101,9 @@ func (cm *ConnectionMaker) InitiateConnections(peers []string, replace bool) []e
...
+		} else if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
```
Comment (peterbourgon) [inline connection_maker.go]: [net.JoinHostPort](https://golang.org/pkg/net/#JoinHostPort)

Comment (bboreham) [inline connection_maker.go]: That would be beyond the scope of this change.  I don't know why it isn't just `peer`.

Comment (peterbourgon) [inline connection_maker.go]: I'm confused. I'm saying use net.JoinHostPort instead of fmt.Sprintf.

File: connection_maker.go (hunk 1 above)
Comment (bboreham) [inline connection_maker.go]: amended

File: connection_maker.go (hunk 2 above)
Comment (bboreham) [inline connection_maker.go]: I know.  (a) the `Sprintf` was there before; cleaning up arbitrary deficiencies is not the intention of this change
(b) I didn't realise `ResolveTCPAddr` needs a port number, so ignore the bit about `peer`.

**Pull request merged**